tools/start-local-vm

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env bash
+
+shopt -s nullglob
+
+arch=${BUILDSYS_ARCH}
+variant=${BUILDSYS_VARIANT}
+vm_mem=4G
+vm_cpus=4
+force_extract=
+declare -A extra_files=()
+
+current_images=()
+boot_image=
+data_image=
+
+readonly repo_root=$(git rev-parse --show-toplevel)
+
+bail() {
+    >&2 echo "$@"
+    exit 1
+}
+
+show_usage() {
+    local error=$1
+    local newline=$'\n'
+    local usage="\
+usage: ${0##*/} [--arch BUILDSYS_ARCH] [--variant BUILDSYS_VARIANT]
+                      [--vm-memory VM_MEMORY] [--vm-cpus VM_CPUS]
+                      [--inject-file LOCAL_PATH[:IMAGE_PATH]]...
+
+Launch a local virtual machine from a Bottlerocket image.
+
+Options:
+
+    --arch              architecture of the Bottlerocket image (must match the
+                        host architecture ($(uname -m)); may be omitted if the
+                        BUILDSYS_ARCH environment variable is set)
+    --variant           Bottlerocket variant to run (may be omitted if the
+                        BUILDSYS_VARIANT environment variable is set)
+    --vm-memory         amount of memory to assign to the VM; VM_MEMORY must be
+                        a valid QEMU memory specifier (default is ${vm_mem})
+    --vm-cpus           number of CPUs to spawn for VM (default is ${vm_cpus})
+    --force-extract     force recreation of the extracted Bottlerocket image,
+                        e.g. to incorporate new --inject-file options
+    --inject-file       adds a local file to the private partition of the
+                        Bottlerocket image before launching the virtual machine
+                        (may be given multiple times)
+    --help              shows this usage text
+
+The virtual machine's port 22 (SSH) will be exposed via the local port 2222,
+i.e. if the Bottlerocket admin container has been enabled via user-data, it can
```

Comment:
For the lack of a better place to ask, what happens to the control container? Does it just fail?

Reply:
I don't know! :-) The control container isn't active in the metal variants, and with, for example, aws-dev early-boot-config fails for lack of access to the IMDS (which likely has knock-on effects on the control container). Here's where the extension idea of ENI pass-through comes into play.